Frontend/Frontend.Client/Pages/Settings.razor

```diff
@@ -0,0 +1,105 @@
+﻿@page "/settings"
+@using Models
+@inject NavigationManager NavigationManager
+
+<PageTitle>Settings - Ban List</PageTitle>
+
+<div class="settings-page-container">
+    <button class="settings-back-button" @onclick="GoBack">BACK</button>
+    <div class="settings-main-block">
+        <div class="settings-main-top">
+            <h1>Ban List</h1>
+        </div>
+        <div class="settings-main-content">
+            <table class="settings-ban-table">
+                <thead>
+                    <tr>
+                        <th>Banned Usernames</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    @foreach (var ban in bans)
+                    {
+                        <tr @onclick="() => ShowBanDetails(ban)">
+                            <td>@ban.Username</td>
+                        </tr>
+                    }
+                </tbody>
+            </table>
+            <table class="settings-ban-table">
+                <thead>
+                    <tr>
+                        <th>Banned IP Addresses</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    @foreach (var ban in bans)
+                    {
+                        <tr @onclick="() => ShowBanDetails(ban)">
+                            <td>@ban.IpAddress</td>
+                        </tr>
+                    }
+                </tbody>
+                </table>
+        </div>
+    </div>
+</div>
+
+@if (showPopup)
+{
+    <div class="ban-popup-overlay">
+        <div class="ban-popup-content">
+            <h2>Ban Details</h2>
+            <p><strong>Username:</strong> @selectedBan.Username</p>
+            <p><strong>IP Address:</strong> @selectedBan.IpAddress</p>
+            <p><strong>Reason:</strong> @selectedBan.Reason</p>
+            <div class="ban-popup-buttons">
```

Comment:
For usability purposes, Add a functionality for hiding the pop up if the user clicks somewhere off the pop up, you can refer the context menu modal for that.

Reply:
Added alongside color unification for the buttons.